favorites.py

```diff
@@ -277,6 +288,48 @@ def add_group(self, s):
 
         self.obj.files["groups"][s] = []
 
+    def prompt_for_alias(self, index, name, group_name=None):
+        """Prompt for an alias for the favorite file.
+
+            Args passed by SelectFavoriteFileCommand:
+
+            index      : the evaluated quick panel index
+            name       : a tuple (alias, path)
+            group_name : the group browsed in the quick panel
+        """
+        def find_index_in(node):
+            '''Find the index of the file that has been just added.'''
+            for f in node:
+                if f['file'] == name:
+                    return index, (f['alias'], name)
+
+        # CASE: index == -1
+        # file has just been added and its index in the sorted json must be
+        # retrieved. argument 'name' is the simple filename in this case, not a
+        # tuple, so we have to address that
+        if index == -1 and group_name:
+            index, name = find_index_in(self.obj.files['groups'][group_name])
+        elif index == -1:
+            index, name = find_index_in(self.obj.files['files'])
+
+        w = sublime.active_window()
+        self.changing_alias_for = index, name, group_name
+        w.show_input_panel("Alias for this file:", name[0],

```

Comment:
The sublime interface was meant to be handled in `favorite_files` not down  in this layer.

Reply:
So I should move these functions in the other file? the problem is that they can be accessed by two different commands (SelectFavorite and AddFavorite)

Reply:
I'll be away for a couple of hours, I'll push the changes later, to be sure I didn't break anything(removed this toggle thing, split the read/write file function). I just wanted to say that about the main thing, that is functions related to alias renaming, right now I can't see a different way than the one here, maybe I can make a new class in favorite_files.py, so both commands can use it, I'll work on it later.